arch/x86/core/fatal.c

```diff
@@ -305,20 +296,20 @@ static void dump_page_fault(z_arch_esf_t *esf)
 	err = esf_get_code(esf);
 	LOG_ERR("Page fault at address %p (error code 0x%lx)", cr2, err);
 
-	if ((err & RSVD) != 0) {
+	if ((err & PF_RSVD) != 0) {
 		LOG_ERR("Reserved bits set in page tables");
 	} else {
-		if ((err & PRESENT) == 0) {
+		if ((err & PF_P) == 0) {
 			LOG_ERR("Linear address not present in page tables");
 		}
 		LOG_ERR("Access violation: %s thread not allowed to %s",
-			(err & US) != 0U ? "user" : "supervisor",
-			(err & ID) != 0U ? "execute" : ((err & WR) != 0U ?
-							"write" :
-							"read"));
-		if ((err & PK) != 0) {
+			(err & PF_US) != 0U ? "user" : "supervisor",
+			(err & PF_ID) != 0U ? "execute" : ((err & PF_WR) != 0U ?
+							   "write" :
+							   "read"));
+		if ((err & PF_PK) != 0) {
 			LOG_ERR("Protection key disallowed");
-		} else if ((err & SGX) != 0) {
+		} else if ((err & PF_SGX) != 0) {
 			LOG_ERR("SGX access control violation");
 		}
```

Comment:
I notice here an opportunity to address a coding guideline violation - 15.7 if else if constructs must be terminated with final else. An empty final else is acceptable. Would you like to do that in this PR, or prefer I can submit it separately?

Reply:
I can submit this ^^ separately, shouldn't hold this patch back. Thanks!